core/src/main/java/org/axonframework/eventsourcing/eventstore/SequenceEventStorageEngine.java

```diff
@@ -92,9 +92,9 @@ public DomainEventStream readEvents(String aggregateIdentifier, long firstSequen
     }
 
     @Override
-    public Optional<DomainEventMessage<?>> readSnapshot(String aggregateIdentifier) {
-        return Optional.ofNullable(activeStorage.readSnapshot(aggregateIdentifier).orElseGet(
-                () -> historicStorage.readSnapshot(aggregateIdentifier).orElse(null)));
+    public Stream<DomainEventMessage<?>> readSnapshots(String aggregateIdentifier) {
+        return Stream.concat(historicStorage.readSnapshots(aggregateIdentifier),
```

Comment:
Question: Aren't you switching the order around of the snapshots? :thinking: 

Initially, first any snapshots from the active storage would be provided, which if empty is replaced by historical snapshots.
Now, you concat _any_ historical snapshots with active snapshots.
So, shouldn't it be active snapshots and then historical snapshtos?

Reply:
Ah, but in regards to your resolve, it makes sense, as you'll want the most recent snapshot.
Cool, think I am good here. Did I miss any assumptions?

Reply:
You're right here. All other engines sort DESC - so, you'll get most recent snapshots first. This should be the case for `SequenceEventStorageEngine`. Pretty good remark!

Reply:
Thanks! :)